tests/console/validate_raid.pm

```diff
@@ -115,20 +106,13 @@ sub prepare_test_data {
             $num_raid_arrays = @raid_arrays;
             @raid            = (($raid_level, $raid0, $raid1), @raid_detail);
         }
-        else {
-            @partitioning = (
-                $raid_partitions_2_arrays, $hard_disks, $linux_raid_member_2_arrays,
-                $btrfs, $swap,
-            );
-            if (is_sle('<=15-SP1') || (is_sle() && get_var('STAGING')) ||
-                is_leap() || (is_tumbleweed() && check_var('FLAVOR', 'NET'))) {
-                # raid0 for swap is required in some products
-                @raid = (($raid_level, $raid0), @raid_detail);
-            }
-            else {
-                @raid = ($raid_both_same_level, @raid_detail);
-            }
-        }
+    }
+    else {
```

Comment:
Besides, you are re-introducing the second false-positive recently fixed by moving `else`, that happens when is `x86_64` and is not SLE <15, for example in openSUSE will not test anything.

Reply:
Perhaps is a good moment to fill in each `if` `else` with right values for `@raid` and `@partitioning` and die in the last `else` that you re-introduced.

Reply:
I've added `elsif (check_var('ARCH', 'x86_64') && is_sle('<15'))` as for now it looks for me like a simplest solution 

VR for SLE12: https://openqa.suse.de/tests/3450402
VR for SLE15+: https://openqa.suse.de/t3450405